src/memory_region.rs

```diff
@@ -5,7 +5,7 @@ use crate::{
     error::EbpfError,
     vm::{Config, ProgramResult},
 };
-use std::{array, cell::UnsafeCell, fmt, ops::Range};
+use std::{array, cell::UnsafeCell, fmt, ops::Range, ptr::copy_nonoverlapping};
```

Comment:
Do we really need the overhead of a runtime memcopy? Can't we use write unaligned pointer with a generic `<T>` instead of a `len` parameter?

Reply:
Ah this is a long story. `write_unaligned` _is_ `copy_nonoverlapping`. At the beginning I did use generics, but then  API wise I preferred the explicit len over a generic, since generics are more error prone when inferred, and when explicit... turbofish. And also for consistency with map().

My hope was that since we always pass a constant value as len, LLVM would be able to propagate it. But since we call this stuff from the jit too, it does make sense to manually ensure that the call is avoided. I kinda hate the api now, but looking at the asm there's no call so success! 😂